streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalProcessorContext.java

```diff
@@ -26,6 +27,9 @@
  */
 public interface InternalProcessorContext extends ProcessorContext {
 
+    @Override
+    StreamsMetricsImpl metrics();
```

Comment:
Override to refine the type from StreamMetrics to StreamMetricsImpl in support of internal usages.